.github/nomad-builder/Dockerfile

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: BUSL-1.1
+
+FROM ubuntu:bionic
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+ARG ARCH="amd64"
+ARG GO_VERSION="1.24.1"
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+ bash \
+ build-essential \
+ ca-certificates \
+ curl \
+ git \
+ xz-utils \
+ zip
+
+# Get Go and get Going ;)
+RUN curl -L https://go.dev/dl/go${GO_VERSION}.linux-${ARCH}.tar.gz | tar -C /opt -zxv
+
+# Best to map the internal user to host user
+# ARG DOCKER_USER=default_user
+# RUN addgroup $DOCKER_USER && useradd -m -g $DOCKER_USER $DOCKER_USER
+# USER $DOCKER_USER
+# ENV PATH="/home/${DOCKER_USER}/go/bin:/opt/go/bin:$PATH"
+
+# FIXME: should be owned by the $DOCKER_USER but for whatever reason we get
+# permission errors on the host system if we do that, even though it's the same s
```

Comment:
note to reviewers: this, and another FIXME note like this in the yaml, is a minor issue that I haven't been able to figure out. I think it's maybe a gid problem? Even if the username is mapped to the host user we can't `mkdir` from within the container in a mounted volume directory. I welcome ideas on why this happens and how to fix it, but I also don't think it's a showstopper.

Reply:
resolved with [80f96d6](https://github.com/hashicorp/nomad/pull/25626/commits/80f96d65d7e557d929740db4c70deb25b0f42253)